### PR TITLE
Switch to nodejs4.3 runtime

### PR DESCRIPTION
**Pull request by kylef**
Node 0.10 is deprecated in AWS Lambda. I don't think any code changes are required, it would seem all the code is Node 4 compatible, but we do not have any tests so I cannot confirm 100%.

### CODE REVIEW

**Comment by pksunkara** [comment]
There's some issues with closing the function. It's the main difference between 0.10 and 4 on lambda. I don't think the change here is enough.

**Comment by w-vi** [comment]
@kylef have you tested it manually? I think `context.callbackWaitsForEmptyEventLoop` might cause problems. Also it should get rid of the `context.succeed()` and others and use callbacks see http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-using-old-runtime.html

**Review by w-vi — DISMISSED** [review]
Needs to be manually tested.

**Comment by pksunkara** [comment]
Basically, this PR is going to be a revert of https://github.com/apiaryio/cloudwatch-to-papertrail/commit/6ea0a36e55ed3497916cd37cb0944423db8174a6?

IIRC, we made that commit because this lambda was hanging without closing.

I think @abtris should definitely review this.

**Comment by w-vi** [comment]
I think we need to use the `context.callbackWaitsForEmptyEventLoop`. How it needs to be setup I am not sure I'd say we need not to wait because otherwise it would hang so setting it to `context.callbackWaitsForEmptyEventLoop = false` should solve the hanging issue.

File: index.js [hunk 1]
```diff
@@ -108,13 +108,13 @@ exports.handler = function (event, context) {
...
       log.close();
-      context.succeed();
+      return cb(null, null);
     }
 
     dogapi.metric.send_all(metricPoints, function () {
       dogapi.metric.send_all(reportPoints, function () {
         log.close();
-        context.succeed();
+        cb(null, null);
```
Comment (abtris) [inline index.js]: nit: better `cb()`

File: index.js [hunk 2]
```diff
@@ -108,13 +108,13 @@ exports.handler = function (event, context) {
 
     if (config.datadog === '') {
       log.close();
-      context.succeed();
+      return cb(null, null);
```
Comment (abtris) [inline index.js]: nit: I think better is `cb()`

**Review by abtris — DISMISSED** [review]
I agree with @w-vi please add `context.callbackWaitsForEmptyEventLoop = false; ` and text manually if doesn't have problems with hang on.

**Comment by kylef** [comment]
> Basically, this PR is going to be a revert of 6ea0a36?

No, before 6ea0a36 there was cases where the callback was not called whereas my changes always explicitly call the callback.

**Comment by kylef** [comment]
Marked as DO NOT MERGE while I manually test these changes.

**Review by w-vi — APPROVED** [review]

**Pull request merged**